static/app/components/events/eventReplay/index.tsx

```diff
@@ -1,26 +1,57 @@
+import {useState} from 'react';
+
 import Button from 'sentry/components/button';
 import ErrorBoundary from 'sentry/components/errorBoundary';
 import EventDataSection from 'sentry/components/events/eventDataSection';
 import LazyLoad from 'sentry/components/lazyLoad';
+import {relativeTimeInMs} from 'sentry/components/replays/utils';
 import {t} from 'sentry/locale';
 
 type Props = {
   orgSlug: string;
   projectSlug: string;
   replayId: string;
+  eventTimestamp?: string;
 };
 
-export default function EventReplay({replayId, orgSlug, projectSlug}: Props) {
+export default function EventReplay({
+  replayId,
+  orgSlug,
+  projectSlug,
+  eventTimestamp,
+}: Props) {
+  const [initialTimeOffset, setInitialTimeOffset] = useState<number | undefined>(
+    undefined
+  );
+
+  const getInitialTimeOffset = startTimestampMs => {
+    if (eventTimestamp && startTimestampMs) {
+      const timeOffset = Math.trunc(
+        relativeTimeInMs(eventTimestamp, startTimestampMs) / 1000 - 5
+      ); // 5 seconds before event
```

Comment:
Lets not truncate the timestamp.

Decimals are important to include. Maybe limit to 3 decimals at most so we get that millisecond granulatiry, if you're worried about  things like 2.3333333333 or something.